src/contrib/cluster/Akka.Cluster.Tools/PublishSubscribe/Query/Protocol.cs

```diff
@@ -0,0 +1,118 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="Messages.cs" company="Akka.NET Project">
+//     Copyright (C) 2013-2023 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System.Collections.Immutable;
+using Akka.Actor;
+
+namespace Akka.Cluster.Tools.PublishSubscribe.Query;
+
+#region Request messages
+
+public sealed class GetLocalPubSubStats
+{
+    public static readonly GetLocalPubSubStats Instance = new();
+    private GetLocalPubSubStats() { }
+}
+
+public sealed class GetPubSubStats
+{
+    public static readonly GetPubSubStats Instance = new();
+    private GetPubSubStats() { }
+}
+
+public sealed class GetClusterPubSubStats
+{
+    public static readonly GetClusterPubSubStats Instance = new();
+    private GetClusterPubSubStats() { }
+}
+
+public sealed class GetLocalPubSubState
```

Comment:
What's the difference between state and stats?

Reply:
Stats will return the subscriber count of each topic, state returns a list of actual subscriber addresses of each topic